src/Symfony/Component/Form/Extension/Core/Type/MultiStepType.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class MultiStepType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setRequired('steps')
+            ->setDefault('current_step', static function (Options $options): string {
```

Comment:
I'd also recommend adding the `setAllowedTypes('current_step', 'string')` method

Reply:
Btw, the default value here might not be useful as the user will need to control it dynamically outside the form type anyway. It might be better to make it a required field instead

Reply:
The User can controll that dynamically from outside. Only when there is nothing set it would be cool to have the default. Thats less implementation for the User.